js_src/dart_helpers.js

```diff
@@ -36,6 +36,34 @@ function _createReactDartComponentClassConfig(dartInteropStatics, componentStati
       return dartInteropStatics.handleRender(this.dartComponent);
     }
   };
+
+  // React limits the accessible context entries
+  // to the keys specified in childContextTypes/contextTypes.
+
+  var childContextKeys = jsConfig && jsConfig.childContextKeys;
+  var contextKeys = jsConfig && jsConfig.contextKeys;
+
+  if (childContextKeys && childContextKeys.length !== 0) {
+    config.childContextTypes = {};
+    for (var i = 0; i < childContextKeys.length; i++) {
+      config.childContextTypes[childContextKeys[i]] = React.PropTypes.object;
+    }
+
+    // Only declare this when `hasChildContext` is true to avoid unnecessarily
```

Comment:
Referencing an old variable?

Reply:
Possibly. I'll dig into it.

Reply:
I think so, sorry about that. Should probably read `when childContextKeys is non-empty`

Reply:
Fixed